src/coreclr/jit/lsraloongarch64.cpp

```diff
@@ -0,0 +1,1343 @@
+                if (objChild->OperGet() == GT_LCL_VAR_ADDR)
+                {
+                    // We will generate all of the code for the GT_PUTARG_STK, the GT_OBJ and the GT_LCL_VAR_ADDR
+                    // as one contained operation, and there are no source registers.
+                    //
+                    assert(objChild->isContained());
+                }
+                else
+                {
+                    // We will generate all of the code for the GT_PUTARG_STK and its child node
+                    // as one contained operation
+                    //
+                    srcCount = BuildOperandUses(objChild);
+                }
+            }
+            else
+            {
+                // No source registers.
+                putArgChild->OperIs(GT_LCL_VAR);
+            }
+        }
+    }
+    else
+    {
+        assert(!putArgChild->isContained());
+        srcCount = BuildOperandUses(putArgChild);
+    }
+    buildInternalRegisterUses();
+    return srcCount;
+}
+
+#if FEATURE_ARG_SPLIT
+//------------------------------------------------------------------------
+// BuildPutArgSplit: Set the NodeInfo for a GT_PUTARG_SPLIT node
+//
+// Arguments:
+//    argNode - a GT_PUTARG_SPLIT node
+//
+// Return Value:
+//    The number of sources consumed by this node.
+//
+// Notes:
+//    Set the child node(s) to be contained
+//
+int LinearScan::BuildPutArgSplit(GenTreePutArgSplit* argNode)
+{
+    int srcCount = 0;
+    assert(argNode->gtOper == GT_PUTARG_SPLIT);
+
+    GenTree* putArgChild = argNode->gtGetOp1();
+
+    // Registers for split argument corresponds to source
+    int dstCount = argNode->gtNumRegs;
+
+    regNumber argReg  = argNode->GetRegNum();
+    regMaskTP argMask = RBM_NONE;
+    for (unsigned i = 0; i < argNode->gtNumRegs; i++)
+    {
+        regNumber thisArgReg = (regNumber)((unsigned)argReg + i);
+        argMask |= genRegMask(thisArgReg);
+        argNode->SetRegNumByIdx(thisArgReg, i);
+    }
+
+    if (putArgChild->OperGet() == GT_FIELD_LIST)
+    {
+        // Generated code:
+        // 1. Consume all of the items in the GT_FIELD_LIST (source)
+        // 2. Store to target slot and move to target registers (destination) from source
+        //
+        unsigned sourceRegCount = 0;
+
+        // To avoid redundant moves, have the argument operand computed in the
+        // register in which the argument is passed to the call.
+
+        for (GenTreeFieldList::Use& use : putArgChild->AsFieldList()->Uses())
+        {
+            GenTree* node = use.GetNode();
+            assert(!node->isContained());
+            // The only multi-reg nodes we should see are OperIsMultiRegOp()
+            assert(!node->IsMultiRegNode());
+
+            // Consume all the registers, setting the appropriate register mask for the ones that
+            // go into registers.
+            for (unsigned regIndex = 0; regIndex < 1; regIndex++)
```

Comment:
Can you remove the single iteration loop?

Reply:
OK, Thanks